wsynphot/spectrum1d.py

```diff
@@ -1,3 +1,82 @@
 import numpy as np
+from astropy import constants as const
 from astropy import units as u
 from starkit.fix_spectrum1d import SKSpectrum1D
+
+
+# TODO: this can eventually be replaced with an astropy3+ function
+# from astropy.modeling import blackbody_lambda
+#
+# this version is adapted from astropy.modeling.blackbody
+def blackbody_lambda(wavelength, temperature):
+    """
+    Calculate the blackbody spectral density per unit wavelength.
+
+    Parameters
+    ----------
+    wavelength : `~astropy.units.Quantity`
+        Wavelength array to evaluate on.
+
+    temperature : `~astropy.units.Quantity`
+        Blackbody temperature.
+    """
+
+    # Convert to units for calculations, also force double precision
+    with u.add_enabled_equivalencies(u.spectral() + u.temperature()):
+        freq = u.Quantity(wavelength, u.Hz, dtype=np.float64)
+        temp = u.Quantity(temperature, u.K, dtype=np.float64)
+
+    log_boltz = const.h * freq / (const.k_B * temp)
+    boltzm1 = np.expm1(log_boltz)
+
+    bb_nu = (2.0 * const.h * freq ** 3 / (const.c ** 2 * boltzm1))
+
+    flam = u.erg / (u.cm**2 * u.s * u.AA)
+    flux = bb_nu.to(flam, u.spectral_density(wavelength))
+
+    return flux / u.sr  # Add per steradian to output flux unit
+
+
+def blackbody1d(temperature, radius, distance=10*u.pc,
+                lambda_min=2000, lambda_max=10000, dlambda=1):
+    """
+    One dimensional blackbody spectrum.
+
+    Parameters
+    ----------
+    temperature : `~astropy.units.Quantity`
+        Blackbody temperature.
+
+    radius : `~astropy.units.Quantity`
+        Radius of spherical blackbody.
+
+    distance : `~astropy.units.Quantity`
+        Distance of blackbody source.
+
+    lambda_min : ~float
+        Minimum wavelength for spectrum.
+
+    lambda_max : ~float
+        Maximum wavelength for spectrum.
+
+    dlambda : ~float
+        Wavelength interval for spectrum.
+
+
+    Returns
+    -------
+    bb : `~starkit.fix_spectrum1d.SKSpectrum1D`
+        Blackbody spectrum.
+    """
+
```

Comment:
`temperature = u.Quantity(temperature, u.K)`

Reply:
This is done in blackbody_lambda